File: maps-samplej/src/main/java/com/omh/android/maps/sample/utils/PermissionsUtils.java
package com.omh.android.maps.sample.utils;

import static com.omh.android.maps.sample.utils.Constants.PERMISSIONS;

import android.content.Context;
import android.content.pm.PackageManager;

import androidx.core.content.ContextCompat;

public class PermissionsUtils {
        public static Boolean grantedRequiredPermissions(Context context) {
            for (String element : PERMISSIONS) {
                if (!(ContextCompat.checkSelfPermission(context, element) == PackageManager.PERMISSION_GRANTED))
                    return false;
            }
            return true;
    }
}

File: maps-sample/src/main/java/com/omh/android/maps/sample/utils/PermissionsUtils.kt
/*
 * Copyright 2023 Open Mobile Hub
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.omh.android.maps.sample.utils

import android.content.Context
import android.content.pm.PackageManager
import androidx.core.content.ContextCompat
import com.omh.android.maps.sample.utils.Constants.PERMISSIONS

object PermissionsUtils {
    fun grantedRequiredPermissions(context: Context): Boolean {
        return PERMISSIONS.all {
            ContextCompat.checkSelfPermission(context, it) == PackageManager.PERMISSION_GRANTED
        }
    }
}
